python/pyarrow/_compute.pyx

```diff
@@ -792,7 +775,7 @@ cdef class _PadOptions(FunctionOptions):
 
 
 class PadOptions(_PadOptions):
-    def __init__(self, width, padding=' '):
+    def __init__(self, width, *, padding=" "):
```

Comment:
Similarly, this doesn't seem desirable. I don't think there's any ambiguity as to what the second argument is.